ksqldb-common/src/main/java/io/confluent/ksql/serde/SerdeOption.java

```diff
@@ -16,29 +16,42 @@
 package io.confluent.ksql.serde;
 
 import com.fasterxml.jackson.annotation.JsonCreator;
-import com.google.common.collect.ImmutableSet;
-import java.util.Set;
+import java.util.Objects;
 
 public enum SerdeOption {
```

Comment:
The mapping between SerdeOption and SerdeFeature always be 1-to-1, right? What's the purpose of having SerdeOption wrap SerdeFeature?

Reply:
My bad, I didn't really explain the intent here!

The PR was raised because the current system, where we only have `UNWRAP_SINGLE_VALUES` as a SerdeOption, isn't enough to express having some formats that don't support wrapping, e.g. `KAFKA`, some that only support wrapping, e.g. `PROTOBUF`, and some that support both.  Hence adding the `WRAP_SINGLE_VALUES` to the enum.

The use of `SerdeFeature` will become apparent in the next PR.  `SerdeFeature`s will be passed to the Serde along with the schema so that they can do what's needed.

> The mapping between SerdeOption and SerdeFeature always be 1-to-1, right? 

No, when we add `WRAP_SINGLE_KEY` and `UNWRAP_SINGLE_KEY` they'll also use `SerdeFeature.WRAP_SINGLES` and `SerdeFeature.UNWRAP_SINGLES`.  (This is needed for KLIP-33, which is why I'm working on it). 

Once this is done, we'll have `SerdeOptions`, which sets at a topic schema level what options are on, and `SerdeFeatures` which control the same at the key/value level. 

In the future it may be possible to do away with `SerdeOption` in favour of just using `SerdeFeature`s tracked at the key and value level, but that's a bigger change, and isn't trivial.

Hope that helps explain the _why_ of this PR.

Reply:
Super helpful. Thanks, Andy!